state/queries.go

```diff
@@ -118,3 +125,43 @@ func (ls *LedgerState) queryShelley(query *olocalstatequery.ShelleyQuery) (any,
 		return nil, fmt.Errorf("unsupported query type: %T", q)
 	}
 }
+
+func (ls *LedgerState) queryShelleyUtxoByAddress(addrs []ledger.Address) (any, error) {
```

Comment:
Can we have better return type than `any`?

Reply:
The parent functions must return `any`, so there's not much point